crates/distribution-types/src/parsed_url.rs

```diff
@@ -22,13 +24,21 @@ pub enum ParsedUrlError {
     UrlParse(String, #[source] url::ParseError),
 }
 
+#[derive(Debug, Clone, Derivative)]
+#[derivative(Hash, PartialEq, Eq)]
+pub struct VerbatimParsedUrl {
+    #[derivative(PartialEq = "ignore", Hash = "ignore")]
```

Comment:
Why is this necessary? Where are we relying on this? I find that this pattern tends to cause a lot of subtle bugs.

Reply:
We were previously relying on equality between two `VerbatimUrl`s. This derive preserve this behavior, so we don't get regression when the `ParsedUrl` is slightly different from the `VerbatimUrl`. Eventually, we should remove that check

Reply:
I think my question is: where are we relying on this? Any way we can make this explicit at those sites?